### PR TITLE
implemented modules for M:D:o365EmailAddresses:mu and GR:D:o365EmailA…

### PR DESCRIPTION
**Pull request by martin-kuba**
…ddresses:mu

Partial implementation for redmine issue 3755 without checking global uniqueness. 
Attribute modules and their unit tests are implemented.

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	private static final String NAMESPACE = AttributesManager.NS_GROUP_RESOURCE_ATTR_DEF;
+	static final String ADNAME_ATTRIBUTE = NAMESPACE + ":adName";
+
+	@Override
+	public void checkAttributeValue(PerunSessionImpl sess, Resource resource, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		List<String> emails;
+		//get values
+		Object value = attribute.getValue();
+		if (value == null) {
+			throw new WrongAttributeValueException(attribute, resource, group, "can't be null.");
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java]: Shouldn't this be checked after check on adName value existence ? What do you think @licehammer ? I'm not sure, what is desired workflow with adName itself - regarding later manual changes and fixes, etc.
Also we translate empty list to null when storing (not sure on retrieving), so you might never get "empty list" on line 73.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	/**
+	 * Prefills value created by joining value of urn:perun:group_resource:attribute-def:def:adName with "@group.muni.cz"
+	 */
+	@Override
+	public Attribute fillAttribute(PerunSessionImpl sess, Resource resource, Group group, AttributeDefinition attrDef) throws InternalErrorException, WrongAttributeAssignmentException {
+		if (!NAMESPACE.equals(attrDef.getNamespace())) throw new WrongAttributeAssignmentException(attrDef);
+		try {
+			Attribute result = new Attribute(attrDef);
+			Object adName =
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java]: Not sure, why splitted to 2 lines ?

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java (hunk 1 above)
Comment (martin-kuba) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java]: OK, I will allow null value representing empty list.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java (hunk 2 above)
Comment (martin-kuba) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java]: just to make it shorter, but can be put on one line.

**Comment by martin-kuba** [comment]
@zlamalp changes implemented

**Pull request merged**